### DISCUSSION TITLE
Chapters: como eles funcionam na sua empresa?

### DISCUSSION THREAD

**Discussion by GusChenn** [frontendbr/forum#2458] [Discussão]
Eai pessoal tudo certo?

Estamos começando a estruturar os chapters aqui na empresa onde trabalho, e o primeira definição que decidimos bater foi **o que é um chapter e o que é esperado de um chapter**.

Li vários depoimentos interessantes na discussão #446, mas essa thread foi aberta alguns anos atrás e fiquei muito curioso pra ver como essa conversa pode ter mudado de alguns anos pra cá.

Dito isso, como funcionam os chapters no lugar onde vocês trabalham? Que tipo de tópicos são discutidos, e eles já produziram que tipo de melhorias/ resultados tanto para a empresa quanto para os desenvolvedores?

Qualquer opinião é bem vinda! O seu depoimento ia ajudar demais a gente!

**Comment by pedrogustavo** [comment]
Olá @GusChenn, aqui na empresa em que trabalho, temos 2 "níveis" de chapter de frontend, e eles funcionam da seguinte forma:

### Chapter Frontend
Contexto: Temos uma única aplicação onde diversos fronts, espalhados em vários times, trabalham, cada um com suas próprias demandas de produto.

Nesse contexto, o chapter geralmente ocorre uma vez por mês, e nele discutimos problemas em comum dessa aplicação específica. Exemplo: padronização do uso de componentes, melhorias no projeto, problemas enfrentados pelos desenvolvedores, etc.
Antes de cada encontro, realizamos uma pesquisa para identificar as principais dificuldades e definir os temas que serão abordados com base nessa pesquisa (utilizamos basicamente um formulário Google).
Durante o encontro, abordamos os tópicos e definimos quem pode implementar as melhorias discutidas. No próximo encontro, fazemos uma revisão do que ficou pendente de ser feito/implementado.

### Chapter Frontend Cross
Contexto: Na companhia em que trabalho, existem várias empresas dentro do grupo, então temos muitos frontends que não têm contato no dia a dia.

Nesse encontro, temos a apresentação de palestras realizadas por qualquer desenvolvedor interessado em compartilhar conhecimento. Durante esse evento, há a disseminação de conteúdos e aprendizados.

Exemplo: No time X, utilizamos a metodologia Y para resolver determinado problema e obtivemos o resultado Z, ou algo técnico de forma mais prática, como "como utilizar tal recurso no React". O tema é livre, e cada um pode compartilhar seus conhecimentos.

Além disso, aproveitamos esse encontro para socializar e trocar ideias com outros desenvolvedores. Esse evento geralmente ocorre com um intervalo de 1 a 2 meses.

Para essas palestras, rodamos um call for papers e quem tiver interesse submete a apresentação que deseja fazer.


Bem, é isso por aqui. Espero ter contribuído com algo!

**Reply by GusChenn** [reply]
Valeu pela contribuição, @pedrogustavo! Aqui também temos várias empresas dentro de nosso grupo, então achei muito interessante essa dinâmica do chapter cross (parece bastante uma [guilda](https://www.qulture.rocks/blog/como-a-spotify-organiza-seus-times-de-produto#:~:text=funcional%20dos%20membros.-,Guilda,-Defini%C3%A7%C3%A3o%20de%20Guilda) no modelo spotify).

**Comment by felipefialho** [comment]
Aqui na empresa temos um Chapter de Front-end que já dura 4 anos, é um pilar importante da nossa cultura de desenvolvimento, esse chapter é organizado por devs do time de plataforma. Mas também temos chapters mais contextuais, focados em produtos específicos, nesse caso devs dos squads organizam conforme necessidade

Vou focar no chapter cross

### Frequência

Funciona quinzenalmente, caso tenha feriados no dia a gente pula essa edição e mantém a outra já marcada

### Temas

Trabalhamos com alguns formatos

- **Tech talk**: Apresentações de temas, tanto de estudos quanto de entregas feitas
- **Papo aberto**:  Discussão mais livre inclusive sobre decisões, onde vamos puxando os temas durante o papo
- **Dojo**:  Pode dar uma olhada no formato nesse [repositório público](https://github.com/juntossomosmais/frontend-dojo)

### Organização

A organização do chapter em si fica com o time de plataforma, mas quem vai apresentar o **tech talk** por exemplo fica como responsável daquela edição

### Extras

Além disso temos um canal async no Slack onde discussões, troca de idéias e dúvidas são incentivadas. Eventualmente alguma discussão iniciada numa thread vira um huddle no próprio canal, isso é legal porque outras pessoas podem participar

**Reply by GusChenn** [reply]
@felipefialho Concordo muito com esse ponto do chapter ser um pilar da cultura de desenvolvimento! Temos essa visão aqui também.

**Comment by GusChenn** [comment]
### Update:

Desde a data da criação dessa thread até agora, nasceu o chapter front end na empresa onde trabalho e já estamos caminhando para concretizar ações super positivas para o time de engenharia de software!

Deixo aqui um resumão da estrutura que adotamos para caso alguém, assim como eu, caia nesse fórum com uma dúvida parecida no futuro.

### Estrutura

Após uma pesquisa sobre a estrutura de chapters de outras empresas e uma análise do que desejávamos nesse momento da empresa, iniciamos o chapter com os seguintes objetivos:
- Quebrar o isolamento entre squads
- Disseminar conhecimento
- Gerar oportunidades de aprendizado
- Manter nossa plataforma atualizada e resiliente
- Gerar inovação

Tendo esses objetivos em mente, organizamos o "roteiro" do encontro síncrono que consiste em uma reunião quinzenal de uma hora onde dividimos o tempo igualmente para:
- Fazer tech talks
- Conversar sobre as implementações recentes de cada squad ressaltando os desafios e aprendizados
- Definir ou acompanhar o progresso de projetos originados como iniciativas do chapter. Esses projetos devem se alinhas aos pilares, e os participantes (a princípio) dedicarão uma hora semanal para essas implementações.
- Conversar sobre pautas coletadas em um formulário prévio ou sobre pautas livres.

Além do momento síncrono, também julgamos importante manter um canal assíncrono de comunicação para compartilharmos artigos, opiniões ou tirar dúvidas técnicas sobre nosso fron end. Qualquer um nesse canal pode puxar discussões desse tipo, mas uma pessoa (a princípio) fixa fica responsável por passar o formulário de tópicos e eventualmente movimentar o canal com novas discussões. Um dos desafios da criação do chapter é fazer com que ele tenha tração com a equipe, e movimentar discussões foi uma forma que encontramos de começar a colocar o chapter nessa posição de "central de conhecimento".

Acho que esses são os pontos mais importantes! Obrigado @pedrogustavo e @felipefialho por contribuírem, e espero que esse texto ajude outras pessoas no futuro também. Se ficar com alguma dúvida, pode chamar!

**Reply by felipefialho** [reply]
Sensacional demais!

**Comment by robleds** [comment]
Olá @GusChenn  e a todos da thread!
Queria compartilhar uma experiência que estou tendo com chapters em um projeto longo de software que envolve 3 a 5 equipes simultâneas e cerca de 20 pessoas. Peguei o conceito de chapters e adaptei para o nosso contexto, com o objetivo de organizar o trabalho, alinhar as práticas e integrar melhor o time.

Dividi os chapters nas seguintes áreas:

- Modelagem de Dados
- Infraestrutura
- Sydle & Sybox
- Integrações e RPA
- Arquitetura dos Processos
- Codificação
- Interfaces e Experiência
- Jornadas de Testes
- Dashboard Analytics

Para cada chapter, coloquei uma ou duas pessoas como líderes, que ajudam a manter a governança, alinhar boas práticas e garantir a interação entre todos. Além disso, as equipes foram organizadas em squads, cada um cuidando de um módulo completo do sistema.

Essa combinação tem sido interessante porque os chapters dão uma visão mais ampla do projeto e promovem a horizontalidade (especialmente em arquitetura), enquanto os squads focam no desenvolvimento dos módulos. Assim, conseguimos manter uma boa integração entre as pessoas, criar uma documentação bem estruturada e reduzir problemas como desalinhamento arquitetural, interfaces desconexas e falhas de usabilidade.

Essa estrutura tem funcionado bem, especialmente porque conseguimos equilibrar o olhar estratégico com a execução prática.

Muito legal ter aberto esse tema e podermos contribuir de alguma forma!
